### PR TITLE
feat: allow custom replay root

### PR DESCRIPTION
**Pull request by daibhin**
## Changes

Add a new `root` option to the JS `session_replay` config

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

**Review by marandaneto — APPROVED** [review]
If that's a quick win and it works with Flutter web, I'd be fine with it.

**Comment by daibhin** [comment]
@pauldambra is there a reason we hadn't included this as an option before?

File: src/extensions/replay/sessionrecording-utils.ts [hunk 1]
```diff
@@ -61,6 +61,7 @@ export declare type recordOptions<T> = {
     plugins?: RecordPlugin[]
     mousemoveWait?: number
     keepIframeSrcFn?: KeepIframeSrcFn
+    root?: HTMLElement
```
Comment (pauldambra) [inline src/extensions/replay/sessionrecording-utils.ts:64]: should we document here why we're adding this config?
do all flutter users have to set it?

File: src/types.ts [hunk 2]
```diff
@@ -171,6 +171,7 @@ export interface SessionRecordingOptions {
     // our settings here only support a subset of those proposed for rrweb's network capture plugin
     recordHeaders?: boolean
     recordBody?: boolean
+    root?: HTMLElement
```
Comment (pauldambra) [inline src/types.ts:174]: oh... or should comment on when and why go here?

**Review by pauldambra — APPROVED** [review]
test failure looks easy to fix

we should definitely confirm this still works for web and solves problem for flutter but otherwise no reason not to expose it - just probably need a really clear comment on when and why to set it

**Comment by daibhin** [comment]
Do not need this in favour of https://github.com/PostHog/posthog-js/pull/1082

**Pull request closed without merging**